iati/tests/test_data.py

```diff
@@ -218,33 +236,48 @@ def test_instantiation_dataset_from_string_with_encoding(self, encoding):
     @pytest.mark.parametrize("encoding_declared, encoding_used", [
         ("UTF-16", "UTF-8"),
         ("UTF-16", "ISO-8859-1"),
+        ("UTF-16", "ASCII"),
         ("UTF-16", "BIG5"),
-        ("UTF-16", "EUC-JP"),
-        ("UTF-32", "UTF-16"),
-        ("ASCII", "UTF-16"),
-        ("ISO-8859-1", "UTF-16"),
-        ("ISO-8859-2", "UTF-16"),
-        ("BIG5", "UTF-16"),
-        ("EUC-JP", "UTF-16")])
-    def test_instantiation_dataset_from_string_with_encoding_mismatch(self, encoding_declared, encoding_used):
+        ("UTF-16", "EUC-JP")
```

Comment:
Not sure why `UTF-32` (amongst others) has been removed here???

Reply:
See: The `Note:` in the docstring:

```
There are a number of other errors that may be raised with alternative encoding mismatches. These are not supported since it does not appear likely enough that they will occur and be a large issue in practice.

This is due to a pair of issues with libxml2 (the underlying library behind lxml):

1. It only supports a limited number of encodings out-of-the-box.
2. Different encoding pairs (whether supported or unsupported by libxml2; byte-equivalent-subsets or distinct encodings; and more), will return different error codes in what one would expect to act as equivalent situations.
```

Reply:
I see!